server/src/main/java/com/cloud/api/ApiServlet.java

```diff
@@ -259,6 +259,22 @@ void processRequestInContext(final HttpServletRequest req, final HttpServletResp
                 userId = (Long)session.getAttribute("userid");
                 final String account = (String) session.getAttribute("account");
                 final Object accountObj = session.getAttribute("accountobj");
+                if (session.getAttribute(ApiConstants.SESSIONKEY) != null) {
+                    Cookie[] cookies = req.getCookies();
+                    if (cookies != null) {
+                        HttpSession finalSession = session;
+                        List<Cookie> sessionKeys = Arrays.stream(cookies).filter(cookie ->  cookie.getName().equals(ApiConstants.SESSIONKEY)
+                                && cookie.getValue().equals(finalSession.getAttribute(ApiConstants.SESSIONKEY))).collect(Collectors.toList());
+                        Cookie validCookie = sessionKeys.get(0);
```

Comment:
Why is the first cookie assumed the valid one? Is the core problem that more than one instance of sessionkey cookie exists?

Reply:
When there are 2 sessions opened (primate and legacy UI) to the same domain, there will be multiple sessionkey cookies